File: tempIdeas/java/src/Pietra.java
public class Pietra extends Oggetto {
    /* parameters inside stats in Oggetto are:
     * spirito
     * attacco
     * durezza
     */

    private String durezza_;

    public Pietra(Integer livello, Integer rarita, Double spirito, Double durezza) {
        super(livello,rarita, spirito);
        durezza_="Durezza";
        insertStat(durezza_, durezza);
        sanitizeInput();
    }

    //METODI DI GET------------------------------
    Double getDurezza() {
        return getValoreStat(durezza_);
    }
    //METODI DI SET------------------------------
    void editDurezza(Double val) {
        incrementStat(durezza_,val);
    }


    public Pietra clone() {
        return new Pietra(getLivello(), getRarita(), getSpirito(), getDurezza());
    }

    public Double ricicla() {
        return calcolaMana()/2.0 + getValoreStat(durezza_)*getRarita();
    }
    public void potenzia(Double mana, String parametro) {
        //da implementare
    }

}

File: tempIdeas/untitled/pietra.cpp
#include "pietra.h"

Pietra::Pietra(int livello, int rarita, double spirito, double durezza) : Oggetto(livello, rarita, spirito), durezza_("Durezza") {
    insertStat(durezza_, durezza);
}

double Pietra::getDurezza() const {
    return getValoreStat(durezza_);
}

Pietra *Pietra::clone() const {
    return new Pietra(*this);
}

void Pietra::editDurezza(double val) {
    incrementStat(durezza_, val);
}

double Pietra::ricicla() const {
    return calcolaMana() / 2 + getValoreStat(durezza_) * getRarita();
}

void Pietra::potenzia(double mana, std::string parametro) {

    mana = sanitizeMana(mana);

    int divisore = 5;
    double incremento = mana / getLivello();

    incrementStat(durezza_, incremento * getRarita() / divisore); //Magia riceve un bonus sicuro oltre alla normale distribuzione

    list<string> statsList = getListaStats();
    if((std::find(statsList.begin(), statsList.end(), parametro) == statsList.end())) {

        incremento = incremento / statsList.size();
        for(auto i = statsList.begin(); i != statsList.end(); i++)
            incrementStat(*i, incremento);
    }
    else {
        incrementStat(parametro, incremento);
    }

    normalizza();
}
